src/Compiler/test/Microsoft.NET.Sdk.Razor.SourceGenerators.Tests/RazorSourceGeneratorTests.cs

```diff
@@ -2512,6 +2491,43 @@ public bool TryResolveAssemblyPaths(CompilationLibrary library, List<string>? as
             }
         }
 
+        private class RazorTest : Verify.Test
+        {
+            public RazorTest([CallerFilePath] string? testFile = null, [CallerMemberName] string? testMethod = null)
+                : base(testFile, testMethod)
+            {
+                // Don't resolve any reference assemblies from NuGet
+                ReferenceAssemblies = new ReferenceAssemblies("custom");
```

Comment:
📝 This line ensures that we don't hit NuGet for anything during testing (unless a specific test overrides this property to a different value).